Introducing file_suffix and file_prefix IOHandler attributes, refs #242

jachym opened the pull request:
# Overview

# Related Issue / Discussion

# Additional Information

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [ ] I'd like to contribute [feature X|bugfix Y|docs|something else] to PyWPS. I confirm that my contributions to PyWPS will be compatible with the PyWPS license guidelines at the time of contribution.
- [ ] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

Commit:
Use format extension as the temp file suffix

jachym committed:
Introducing file_suffix and file_prefix IOHandler attributes, refs #242

jachym commented:
@elemoine what do you think about something like this?

elemoine commented:
@jachym did you see https://github.com/geopython/pywps/compare/develop...elemoine:suffix2 (mentioned in https://github.com/geopython/pywps/pull/243)? It reiles on the extension set on the `Format` object instead of introducing new options.

Also, in your patch, how does the user set `file_suffix` and `file_prefix`? It seems to me that the higher-level classes should be changed as well, no?

jachym commented:
@elemoine my pull is based on suffix2 branch of yours (yes, rewritten a bit). 

instead of using `hasattr`, I've rewritten `data_format` method and added the format extension there automatically - in that way, the suffix is set automatically by the format, but can be re-set any time later using the `file_suffix` input object attribute.

Do you have any major objections or are you seeing major drawbacks?

elemoine commented:
I'll have a closer look tomorrow. Thanks a lot @jachym!

elemoine commented on pywps/inout/basic.py:
```diff
@@ -535,7 +537,7 @@ class ComplexInput(BasicIO, BasicComplex, IOHandler):
     def __init__(self, identifier, title=None, abstract=None,
                  workdir=None, data_format=None, supported_formats=None,
-                 mode=MODE.NONE):
+                 mode=MODE.NONE, file_suffix=None):
```
I don't understand the purpose and effect of this new constructor argument.

elemoine commented:
> but can be re-set any time later using the file_suffix input object attribute

Do you mean that the user is able to specify the `file_suffix` and override the format extension? My understanding of your patch is that the format extension is the one with the higher priority right now. No?
